src/line/funcs/functions.js

```diff
@@ -0,0 +1,80 @@
+/**
+ * @jest-environment jsdom
+ */
+const fs = require('fs');
+require('@testing-library/jest-dom');
+const domTesting = require('@testing-library/dom');
+const userEvent = require('@testing-library/user-event').default;
+
+// function used for initialized dom from files. got this function from lecture
+function initDomFromFiles(htmlPath, jsPath) {
+	const html = fs.readFileSync(htmlPath, 'utf8');
+	document.open();
+	document.write(html);
+	document.close();
+
+	require(jsPath)
+}
+
+// this function will init input fields and buttons that we need to use for
+// testing. this will save us a few lines of code in the long run.
+function initElements(elements){
+
+	elements.X_inputs = domTesting.getAllByLabelText(document, 'X');
+	elements.Y_inputs = domTesting.getAllByLabelText(document, 'Y');
+	elements.X_label = domTesting.getByLabelText(document, 'X label');
+	elements.Y_label = domTesting.getByLabelText(document, 'Y label');
+	elements.title_input = domTesting.getByLabelText(document, 'Chart title');
+	elements.color_input = domTesting.getByLabelText(document, 'Chart color');
+	elements.add_input_button = domTesting.getByText(document, '+');
+	elements.generate_chart_button = domTesting.getByText(document, 
+														  'Generate chart');
+	elements.clear_chart_button = domTesting.getByText(document, 
+													   'Clear chart data');
+
+}
+
```

Comment:
It might be simpler and easier to read and understand if you returned the json object from this function. It is hard to determine where elements comes from, but just returning elements in a json could be simpler. However, if there's something I'm missing, feel free to let me know.